### ISSUE TITLE
Assertion failing in Ember beta/canary attempted to remove a function listener

### ISSUE THREAD

**Issue by jrjohnson** [ember-intl/ember-intl#1112]
- [x] I am on the [latest](https://github.com/ember-intl/ember-intl/releases/latest) ember-intl version
- [x] I have searched the [issues](https://github.com/ember-intl/ember-intl/issues) of this repo and believe that this is not a duplicate

Tests in my addon are failing against beta/canary with the message
`Error: Assertion Failed: You attempted to remove a function listener which did not exist on the instance, which means you may have attempted to remove it before it was added.` wheverever the `{{t}}` helper is used. Since the same tests are failing on `master` here I didn't dig too much further just opening this issue for discussion and for others to find.

### Steps to Reproduce

Run `ember-intl` tests against beta/canary for example: 
https://travis-ci.org/ember-intl/ember-intl/jobs/639709922#L605

**Comment by romgere** [comment]
I'm encountering the same error.

`ember-beta` tests failed with `You attempted to remove a function listener which did not exist on the instance, which means you may have attempted to remove it before it was added`

After some investigations, it seems the error is throw by this line  https://github.com/ember-intl/ember-intl/blob/master/addon/helpers/-format-base.js#L46

**Comment by jasonmit** [comment]
> In my case, it's an integration test of a component. The component include 4 calls to {{t}} helper & when I'm debugging, the init method of helpers/-format-base.js is called 4 times & the willDestroy 5 times 🤔

Feels like an Ember bug.  If it goes unpatched upstream, I'll introduce a guard around the `off`.

**Comment by jrjohnson** [comment]
@jasonmit sorry, I can add more details if you need them, I just assumed if there were failing test cases in this library's own suite that they would be the best way to verify. Would building a reproduction app be helpful?

**Comment by jasonmit** [comment]
@jrjohnson I generally allow beta and canary channels to fail.  I find upstream issues breaking the build too often.  I definitely research further if the bug continues with Ember stable.

`willDestroy` being called an extra time seems dubious.  It may be worth creating an Ember.js issue since this part of the lib triggering this exception hasn't changed in awhile.

**Comment by jrjohnson** [comment]
Thanks @jasonmit, I've created the issue linked here.

**Comment by kturney** [comment]
FWIW, I encountered this error while working on https://github.com/ember-intl/ember-intl/pull/1106.
The fix is https://github.com/ember-intl/ember-intl/pull/1106/commits/d43a1f95a6e394732897de254b2f640ab9a268eb and the error comes from [this deprecation](https://deprecations.emberjs.com/v3.x/#toc_events-inherited-function-listeners) being switched to an error [here](https://github.com/emberjs/ember.js/commit/f21955e0ff1b9f6eb14c9444b26078afbc52569e#diff-bf6ea11a7c025a4430907ebb36d974dfL582-R538).
I can port the fix over if you like.

**Comment by jasonmit** [comment]
Yes please @kturney.  Thanks a ton!  If you can land on the 4.x branch since that's stable/latest.  I'll cherry pick into master as well.

**Comment by kturney** [comment]
@jasonmit right now I'm getting
```
Cannot find module '@ember-data/canary-features/addon/default-features.js'
Require stack:
- /Users/kyle.turney/dev/ember-intl/node_modules/@ember-data/-build-infra/src/features.js
- /Users/kyle.turney/dev/ember-intl/node_modules/@ember-data/-build-infra/src/debug-macros.js
- /Users/kyle.turney/dev/ember-intl/node_modules/@ember-data/-build-infra/src/addon-build-config-for-data-package.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-data/index.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/models/package-info-cache/package-info.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/models/package-info-cache/index.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/models/project.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/utilities/get-config.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/utilities/instrumentation.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/lib/cli/index.js
- /Users/kyle.turney/dev/ember-intl/node_modules/ember-cli/bin/ember
```
while trying to run the ember-canary scenario locally.
Would bringing https://github.com/ember-intl/ember-intl/pull/1042 into the 4.x branch fix that?

**Comment by jasonmit** [comment]
I wish I knew of a better solution but cherry picking that over is fine.  Frankly was exhausted from dealing with 2.12 breaks.

Feel free to cherry pick that into your PR.

**Comment by jasonmit** [comment]
Resolved in 4.2.3 thanks to @kturney !

**Comment by jrjohnson** [comment]
Thanks @kturney!
